lib/intercom/search_collection_proxy.rb

```diff
@@ -0,0 +1,82 @@
+require "intercom/utils"
+
+module Intercom
+  class SearchCollectionProxy
+
+    attr_reader :resource_name, :resource_class
+
+    def initialize(resource_name, search_details: {}, client:)
+      @resource_name = resource_name
+      @resource_class = Utils.constantize_resource_name(resource_name)
+      @search_url = search_details[:url]
+      @search_params = search_details[:params]
+      @client = client
+    end
+
+    def each(&block)
+      loop do
+        response_hash = @client.post(@search_url, payload)
+        raise Intercom::HttpError.new('Http Error - No response entity returned') unless response_hash
+        deserialize_response_hash(response_hash, block)
+        break unless has_next_link?(response_hash)
+      end
+      self
+    end
+
+    def [](target_index)
```

Comment:
Is this jumping to a specific element in the search? Not sure I understand this 😅

Reply:
Yes, that's been used in `ClientCollectionProxy` and `ScrollCollectionProxy` similarly

Reply:
Cool! If it's elsewhere then that's cool 😄